### PR TITLE
fix(pipeline): support for SDK changes

### PR DESCRIPTION
**Pull request by jbrockopp**
Dependent on https://github.com/go-vela/sdk-go/pull/161

Part of the effort for https://github.com/go-vela/community/issues/460

This change updates the CLI to support the new and updated SDK functions for `pipelines`.

A `vela get pipelines` command has been added which displays a list of pipelines for a repo in table format:

```shell
$ release/darwin/amd64/vela get pipelines --org JordanBrockopp --repo vela-play
COMMIT                                  	REF             	TYPE	VERSION	STAGES	STEPS
379012cab91195b19b6e8f98932d7d74e10ab002	refs/heads/sleep	yaml	1      	false 	true
```

Like other `vela get` commands, the `output` flag can be used to customize the returned output:

```shell
$ release/darwin/amd64/vela get pipelines --org JordanBrockopp --repo vela-play --output wide
COMMIT                                  	REF             	TYPE	VERSION	EXTERNAL_SECRETS	INTERNAL_SECRETS	SERVICES	STAGES	STEPS	TEMPLATES
379012cab91195b19b6e8f98932d7d74e10ab002	refs/heads/sleep	yaml	1      	false           	false           	false   	false 	true 	false
```

The `vela view pipeline` command still returns the raw pipeline unless the `output` flag is provided:

```shell
$ release/darwin/amd64/vela view pipeline --org JordanBrockopp --repo vela-play --ref 379012cab91195b19b6e8f98932d7d74e10ab002
version: "1"

steps:
  - name: sleep
    image: alpine:latest
    commands:
      - sleep 10
      - echo "10 seconds"
      - sleep 10
      - echo "10 seconds v2"
      - sleep 10
      - echo "10 seconds v3"
      - sleep 10
      - echo "10 seconds v4"
      - sleep 10
      - echo "10 seconds v5"
```

The existing `vela compile pipeline` and `vela expand pipeline` commands have been updated to require the `ref` flag.

### CODE REVIEW

File: action/pipeline/get_test.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+				Repo:    "octocat",
+				Page:    1,
+				PerPage: 10,
+				Output:  "yaml",
+			},
+		},
+	}
+
+	// run tests
+	for _, test := range tests {
```
Comment (cognifloyd) [inline action/pipeline/get_test.go:89]: Would you please name the tests and use `t.Run` to turn these into subtests?

File: action/pipeline/table_test.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+			failure: false,
+			pipelines: &[]library.Pipeline{
+				*p1,
+				*p2,
+			},
+		},
+	}
+
+	// run tests
+	for _, test := range tests {
```
Comment (cognifloyd) [inline action/pipeline/table_test.go:36]: `t.Run`

File: action/pipeline/table_test.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
```
Comment (cognifloyd) [inline action/pipeline/table_test.go:76]: `t.Run`

File: command/pipeline/get_test.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+			set:     authSet,
+		},
+		{
+			failure: true,
+			set:     flag.NewFlagSet("test", 0),
+		},
+	}
+
+	// run tests
+	for _, test := range tests {
```
Comment (cognifloyd) [inline command/pipeline/get_test.go:58]: `t.Run`

**Review by cognifloyd — CHANGES_REQUESTED** [review]
For all the new test files, would you please name the tests and use `t.Run()` so they show up as subtests?
Refactoring the rest of the tests can go in later PRs, but naming things is much easier when adding the tests initially. :)

File: command/pipeline/get_test.go (hunk 4 above)
Comment (jbrockopp) [reply command/pipeline/get_test.go:62]: fixed 👍

File: action/pipeline/table_test.go (hunk 3 above)
Comment (jbrockopp) [reply action/pipeline/table_test.go:82]: fixed 👍

File: action/pipeline/table_test.go (hunk 2 above)
Comment (jbrockopp) [reply action/pipeline/table_test.go:38]: fixed 👍

File: action/pipeline/get_test.go (hunk 1 above)
Comment (jbrockopp) [reply action/pipeline/get_test.go:95]: fixed 👍

**Review by cognifloyd — APPROVED** [review]
Thanks! LGTM

**Review by wass3r — APPROVED** [review]

**Pull request merged**